test/bin/vault/test_usage.py

```diff
@@ -210,4 +210,42 @@ def test_stash_exception_view(self):
             args= parse_args(["archive", "--stash"])
         except:
             success = True
-        self.assertTrue(success)
\ No newline at end of file
+        self.assertTrue(success)
+
+    def test_file_exception_fofn(self):
+        success = False
+        try:
+            args= parse_args(["archive", "--stash", "/file1", "/file2", "--fofn" "/file3"])
+        except:
+            success = True
+        self.assertTrue(success)
+
+        success = False
+        try:
+            args= parse_args(["archive", "--fofn" "/file3", "/file1", "/file2"])
+        except:
+            success = True
+        self.assertTrue(success)
+
+        success = False
+        try:
+            args= parse_args(["keep", "--fofn" "/file3", "/file1", "/file2"])
+        except:
+            success = True
+        self.assertTrue(success)
+
+        success = False
+        try:
+            args= parse_args(["recover", "--fofn" "/file3", "/file1", "/file2"])
+        except:
+            success = True
+        self.assertTrue(success)
+
+        success = False
+        try:
+            args= parse_args(["untrack", "--fofn" "/file3", "/file1", "/file2"])
+        except:
```

Comment:
all throughout here we using generic `except`s - we should instead be ensuring we capture the exception we'd expect, because:
a) if we get a different exception to what we expect, we want to know and not just have it pass the test
b) if we want to interrupt the tests, a `KeyboardInterrupt` would be caught by these, and just pass the test and not interrupt it

Also, we should use assertRaises, i.e. 
```python
self.assertRaises(ArgumentError, parse_args(["archive", "--stash", "/file1", "/file2", "--fofn", "/file3"]))
```

Reply:
Good point, yes - changing this.